tt_eager/tt_dnn/op_library/complex/complex_ops.cpp

```diff
@@ -233,6 +233,62 @@ ComplexTensor polar(const ComplexTensor& input, const MemoryConfig& output_mem_c
     return ComplexTensor({r,i});
 }
 
+// backward ops for type2 complex tensor
+// complex conj type2
+std::vector<ComplexTensor> conj_bw(const ComplexTensor& grad, const ComplexTensor& input, const MemoryConfig& output_mem_config) {
+    std::vector<ComplexTensor> grad_tensor;
+    ComplexTensor grad_result = conj(grad, output_mem_config);
+    grad_tensor.emplace_back(grad_result);
+    return grad_tensor;
+}
+
+// complex imag
+// imag: at::imag(grad)
+std::vector<ComplexTensor> imag_bw(const Tensor& grad, const ComplexTensor& input, const MemoryConfig& output_mem_config) {
+    std::vector<ComplexTensor> grad_tensor;
+    Tensor r = zeros_like(real(input, output_mem_config), output_mem_config);
+    ComplexTensor grad_result = ComplexTensor({r,grad});
+    r.deallocate();
+    grad_tensor.emplace_back(grad_result);
+    return grad_tensor;
+}
+
+// complex real
+// real: at::real(grad)
+std::vector<ComplexTensor> real_bw(const Tensor& grad, const ComplexTensor& input, const MemoryConfig& output_mem_config) {
+    std::vector<ComplexTensor> grad_tensor;
+    Tensor i = zeros_like(real(input, output_mem_config), output_mem_config);
+    ComplexTensor grad_result = ComplexTensor({grad, i});
+    i.deallocate();
+    grad_tensor.emplace_back(grad_result);
+    return grad_tensor;
+}
+
+// complex add
+// self: grad, other: grad * alpha
+std::vector<ComplexTensor> complex_add_bw(const ComplexTensor& grad, const ComplexTensor& input, const ComplexTensor& other, float alpha, const MemoryConfig& output_mem_config) {
+    std::vector<ComplexTensor> grad_tensor;
+    ComplexTensor grad_a = grad;
+    grad_tensor.emplace_back(grad_a);
+    const Tensor& grad_r = grad.real();
+    const Tensor& grad_i = grad.imag();
+    ComplexTensor grad_b = ComplexTensor({mul_unary(grad_r, alpha, output_mem_config), mul_unary(grad_i, alpha, output_mem_config)});
+    grad_tensor.emplace_back(grad_b);
+    return grad_tensor;
+}
+
+// complex sub
+// self: grad, other: -grad * alpha
+std::vector<ComplexTensor> complex_sub_bw(const ComplexTensor& grad, const ComplexTensor& input, const ComplexTensor& other, float alpha, const MemoryConfig& output_mem_config) {
+    std::vector<ComplexTensor> grad_tensor;
+    ComplexTensor grad_a = grad;
+    grad_tensor.emplace_back(grad_a);
+    const Tensor& grad_r = grad.real();
+    const Tensor& grad_i = grad.imag();
+    ComplexTensor grad_b = ComplexTensor({mul_unary(neg(grad_r, output_mem_config), alpha, output_mem_config), mul_unary(neg(grad_i, output_mem_config), alpha, output_mem_config)});
```

Comment:
You can use unary_chain here

Reply:
I've implemented using unary_chain now. thanks